doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,218 @@
+provided power source to the node. It will involve sensing of environmental
+conditions such as temperature, air quality and composition, and be able to
+feed back data at a rate that allows control of those conditions.
+
+
+# 3. List of design philosophies
+
+Below are a list of our design philosophies, including, where appropriate,
+an indication of relative priorities.
+
+## Energy efficiency
+
+RIOT nodes should be able to survive for years without running out of power.
+This means that they need to be able to be extremely conservative with power,
+and harvest any available energy from the environment.
+
+## Real-time capabilities
+
+For non-control and low-frequency sensing applications, precise knowledge of the
+data capture time is not neccessary. Nevertheless, RIOT users may wish to use
+the operating system in control or high frequency data capture applications, and
+in such a case, harder real-time guarantees and more accurate timing should be
+achievable at the expense of power consumption and configuration difficulty.
+
+## Small memory footprint
+
+RIOT aims to support constrained devices in class 1 [1]. This means that
+RIOT applications, including their libraries, consume ~ 10 KiB of RAM and ~100
+KiB of ROM, or less. All modules provided by RIOT can be used practically in
+programs of such sizes, except in exceptional circumstances.
+
+## Constrained networks
+
+Communication is typically constrained in situations where nodes are:
+
+- Distant from each other or a gateway
+- Low transmission power and separated by physical barriers such as walls.
+
+This requires physical layer protocols that can cope with high loss and low
+bandwidth.
+
+Additionally, nodes are low resource and cannot support traditional TCP/IP stack
+protocols. RIOT therefore supports emerging protocols that are the best
+available to handle these challenging conditions, as well as as a validation
+platform for them.
+
+## Short learning curve
+
+Users of RIOT want to be able to develop and deploy their products in as short a
+time as possible, with as little extra learning and specialized knowledge as
+possible. This means support for standards such as POSIX and a range of
+widely-used languages, out-of-the-box examples, simple build configuration, and
+tools and platforms that support RIOT.
+
+However, RIOT's reputation is tied to the robustness of those products, and the
+users should never be able to cut corners in such a way that the products are
+not production ready, even given tight resource constraints. 
+
+Developers, likewise, should be able to start contributing in as short a time as
+possible, and so the design and APIs should remain clear and generally follow
+common and easy to understand principles, unless there are strong benefits that
+directly achieve operational goals.
+
+## Security
+
+Security in all cases is non-negotiable and overrides any competing design
+priorities. This is because of the high importance of protecting users' privacy,
```

Comment:
Security is not "user's privacy". This para needs rethinking, I suppose.

Reply:
I have rewritten the first paragraph - is it more correct?